internal/core/item_tree.rs

```diff
@@ -286,6 +286,108 @@ pub fn visit_item_tree<Base>(
     }
 }
 
+// Some helper to navigate the item_tree:
+
+/// This is a local item tree of one component with some helpful functions to navigate the tree.
+pub struct ItemTree<'a, T> {
+    item_tree: &'a [ItemTreeNode<T>],
+}
+
+impl<'a, T> ItemTree<'a, T> {
+    /// Cretae a new `ItemTree` from its raw data.
+    pub fn new(item_tree: &'a [ItemTreeNode<T>]) -> Self {
+        Self { item_tree }
+    }
+
+    /// Get the parent of a node, returns `None` if this is the root node of this item tree.
+    pub fn parent(&self, index: usize) -> Option<usize> {
+        if index >= self.item_tree.len() || index == 0 {
+            None
+        } else {
+            match &self.item_tree[index] {
+                ItemTreeNode::Item { parent_index, .. } => Some(*parent_index as _),
+                ItemTreeNode::DynamicTree { parent_index, .. } => Some(*parent_index as _),
+            }
+        }
+    }
+
+    /// Returns the next sibling or `None` if this is the last sibling.
+    pub fn next_sibling(&self, index: usize) -> Option<usize> {
+        if let Some(parent_index) = self.parent(index) {
+            match self.item_tree[parent_index] {
+                ItemTreeNode::Item { children_index, children_count, .. } => {
+                    if index >= (children_count as usize + children_index as usize - 1) {
+                        None
+                    } else {
+                        Some(index + 1)
+                    }
+                }
+                ItemTreeNode::DynamicTree { .. } => {
+                    unreachable!("Parent in same itemtree is a repeater.")
+                }
+            }
+        } else {
+            None // No parent, so we have no siblings either:-)
+        }
+    }
+
+    /// Returns the previous sibling or `None` if this is the first sibling.
+    pub fn previous_sibling(&self, index: usize) -> Option<usize> {
+        if let Some(parent_index) = self.parent(index) {
+            match self.item_tree[parent_index] {
+                ItemTreeNode::Item { children_count, .. } => {
+                    if index > children_count as usize {
```

Comment:
This looks wrong to me.
This shouldn't depends on children_count at all.
Maybe you mean the chioldren_index?

Reply:
I think you are right! Thanks for catching this!